test/NuGet.Core.Tests/NuGet.Packaging.Test/ContentModelTests/PatternExpressionMatchTests.cs

```diff
@@ -0,0 +1,198 @@
+                {"type", new ContentPropertyDefinition("text") }
+            };
+            var pattern = new PatternDefinition("content/file.txt");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file.txt";
+            var result = expression.Match(path, propertyDefinitions);
+            Assert.NotNull(result);
+            Assert.Equal(path, result.Path);
+        }
+
+        [Fact]
+        public void LiteralSegment_OnlyShouldNotMatch()
+        {
+            // should not match the literal segment patern because file.txt != file2.txt
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions =
+                new Dictionary<string, ContentPropertyDefinition>() {
+                {"type", new ContentPropertyDefinition("text") }
+            };
+            var pattern = new PatternDefinition("content/file.txt");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file2.txt";
+            var result = expression.Match(path, propertyDefinitions);
+            Assert.Null(result);
+        }
+
+        [Fact]
+        public void LiteralSegment_hasTokenSegmentShouldMatch()
+        {
+            //We have a parser for the token "name" which takes in any type of string and returns it.
+            //Therefore file.txt should be added to the result content item
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions =
+                new Dictionary<string, ContentPropertyDefinition>() {
+                {"name", new ContentPropertyDefinition("any", parser: (o, t) => o) }
+            };
+            //this pattern has both a token segment and a litteral segment
+            var pattern = new PatternDefinition("content/{name}");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file.txt";
+            var match_result = expression.Match(path, propertyDefinitions);
+            object result;
+            match_result.Properties.TryGetValue("name", out result);
+            Assert.NotNull(result);
+            Assert.Equal("file.txt", result);
+        }
+
+        [Fact]
+        public void LiteralSegment_hasTokenSegmentShouldNotMatch()
+        {
+            // We do not have a parser for the token segment name.
+            //As a result it should not be able to match and return a content item
+            Dictionary<string, ContentPropertyDefinition> propertyDefinitions =
+                new Dictionary<string, ContentPropertyDefinition>() {
+                {"name", new ContentPropertyDefinition("any") }
+            };
+            //this pattern has both a token segment and a litteral segment
+            var pattern = new PatternDefinition("content/{name}");
+            var expression = new PatternExpression(pattern);
+            var path = "content/file.txt";
+            var match_result = expression.Match(path, propertyDefinitions);
```

Comment:
I think for locals, we stick to camel casing. Is that right @nkolev92 ?

```suggestion
            var matchResult = expression.Match(path, propertyDefinitions);
```